hack/helm/release/readme.md

```diff
@@ -0,0 +1,91 @@
+
+1. A single ECK Helm chart (eck-operator)
+2. Multiple ECK Helm charts from a directory (eck resources Helm charts), potentially excluding any named Helm charts.
+
+It is also designed to handle releasing to both production, and non-production Helm Chart repositories.
+
+## Buildkite integration
+
+
+### Releasing ECK Operator Helm Chart
+
+When a semver compatible tag is pushed to git (as during a normal release), the following process will be triggered in Buildkite
+
+1. Release tool will be run with `--charts-dir` pointing directly to `eck-operator` chart directory (ignoring all eck resource Helm charts), and set to release to dev bucket/repo.
+2. If successful, the same operation will be run pointing to production bucket/repo.
+
+#### Manually releasing the ECK Operator Helm Chart
+
+To manually publish the ECK Operator Helm chart and it's associated CRD chart, run the below `curl` command.
+
+*Note if there are changes to the Helm tooling that are still in a PR status, you'll need to add "pull_request_repository":"git://github.com/your-username/cloud-on-k8s","pull_request_id":"pr-id" to the below API call.*
+
+```
+  curl --request POST --url https://api.buildkite.com/v2/organizations/elastic/pipelines/cloud-on-k8s-operator/builds \
+    --header 'Authorization: Bearer your-bk-token' --header 'Content-Type: application/json' \
+    --data '{
+        "commit": "HEAD",
+        "branch": "main",
+        "env": {
+          "HELM_DRY_RUN": "false",
+          "HELM_BRANCH": "2.6"
+        },
+        "message": "release eck-operator helm chart"
+    }'
+```
+
+You can then track the progress of the build in [Buildkite](https://buildkite.com/elastic/cloud-on-k8s-operator)
+
+### Releasing ECK Resources Helm Charts
+
+When a commit is merged to `main` branch, which includes any change to a `*/Chart.yaml` file (detectable via `git diff --name-only HEAD~1`)
+
+1. Release tool will be run with `--charts-dir` pointing directly to `deploy` chart directory (containing all charts, including `eck-operator`), set to release to dev bucket/repo, and with `--excludes` flag set to `eck-operator`.
+2. If successful, the same operation will be run pointing to production bucket/repo.
+
+#### Manually releasing the ECK Resources Helm Charts
+
+To manually publish the ECK Resources Helm charts, run the below `curl` command.
+
+*Note if there are changes to the Helm tooling that are still in a PR status, you'll need to add "pull_request_repository":"git://github.com/your-username/cloud-on-k8s","pull_request_id":"pr-id" to the below API call.*
+
+```
+  curl --request POST --url https://api.buildkite.com/v2/organizations/elastic/pipelines/cloud-on-k8s-operator/builds \
+    --header 'Authorization: Bearer your-bk-token' --header 'Content-Type: application/json' \
+    --data '{
+        "commit": "HEAD",
+        "branch": "main",
+        "env": {
+          "HELM_DRY_RUN": "false",
+          "HELM_BRANCH": "2.6"
```

Comment:
What the interest of using this `HELM_BRANCH` environment variable instead of the regular required properties `commit` and `branch`?

Reply:
@thbkrkr readme updated to try and explain this.  Let me know if it's more clear.

Reply:
`branch` and `commit` properties select the branch from which to build the tool and `HELM_BRANCH` the branch from which to release? It's not clear for me why we couldn't just use `branch` and `commit` for the two. IIUC one interest to separate the two is to be able to release code where the tool didn't exist?

Reply:
I see your point. I guess the only possible use case for this after merge of this PR would be if changes were needed to the tooling. But if that was the case then we could just merge the tooling changes prior to running the tooling. I will simply remove this to make things more clear.